libraries/BluetoothSerial/src/BluetoothSerial.cpp

```diff
@@ -445,4 +662,128 @@ esp_err_t BluetoothSerial::register_callback(esp_spp_cb_t * callback)
     return ESP_OK;
 }
 
+//Simple Secure Pairing
+void BluetoothSerial::enableSSP() {
+    _enableSSP = true;
+}
+/*
+     * Set default parameters for Legacy Pairing
+     * Use fixed pin code
+*/
+bool BluetoothSerial::setPin(const char *pin) {
+    bool isEmpty =  !(pin  && *pin);
+    if (isEmpty && !_isPinSet) {
+        return true; // nothing to do
+    } else if (!isEmpty){
+        _pin_len = strlen(pin);
+        memcpy(_pin_code, pin, _pin_len);
+    } else {
+        _pin_len = 0; // resetting pin to none (default)
+    }
+    _isPinSet = true;
```

Comment:
you should `_pin_code[_pin_len] = 0;` so that the pin can be printed easier later.

Reply:
corrected